### PR TITLE
Construct pushURL from remote instead of hardcoding github.com

### PR DESCRIPTION
**Pull request by rdpa**

### CODE REVIEW

**Comment by rdpa** [comment]
closes #91

File: pkg/git/git.go [hunk 1]
```diff
@@ -68,6 +69,18 @@ func (g *Git) Push(workingDir string, args ...string) error {
...
 }
 
+// GetPushURL returns the push url with a token inserted
+func (g *Git) GetPushURL(remote string, token string) (string, error) {
+	pushURL, err := exec.Command("git", "remote", "get-url", "--push", remote).Output()
+	if err != nil {
+		return "", err
+	}
+
+	pushURLArray := strings.SplitAfter(strings.TrimSpace(string(pushURL)), "https://")
```
Comment (unguiculus) [inline pkg/git/git.go:79]: This probably won't work for SSH URLs. Please add a test that verifies the functionality for both, SSH and HTTPS URLs.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Sorry, this slipped through.

**Comment by rdpa** [comment]
Thanks for the review @unguiculus. I will pick this back up soon. FWIW, the original functionality did not work with SSH either, but regardless it would be good to support, and I should add tests.

File: pkg/git/git.go (hunk 1 above)
Comment (unguiculus) [reply pkg/git/git.go:79]: You're right regarding SSH. That's not supported and a token is needed for pushing. chart-releaser is usually used with GitHub Actions which don't use SSH anyways.

**Comment by afritzler** [comment]
Any updates on this one? Having the same issue here.

**Comment by rdpa** [comment]
@unguiculus tests added

**Review by cpanato — APPROVED** [review]
lgtm this change, dont see big issue

**Review by unguiculus — APPROVED** [review]

**Pull request merged**